common/xbps-src/shutils/chroot.sh

```diff
@@ -106,11 +106,30 @@ chroot_prepare() {
         msg_error "Bootstrap not installed in $XBPS_MASTERDIR, can't continue.\n"
     fi
 
-    # Create some required files.
-    if [ -f /etc/localtime ]; then
-        cp -f /etc/localtime $XBPS_MASTERDIR/etc
+    # Setup timezone information
+    #
+    # According to https://www.freedesktop.org/software/systemd/man/localtime.html:
+    #
+    #   Because the timezone identifier is extracted from the symlink target
+    #   name of /etc/localtime, this file may not be a normal file or hardlink.
+    #
+    # So we cannot /etc/localtime into $XBPS_MASTERDIR. Instead, copy over 
+    # the target of the link, creating any required directories along the way.
+    local tzfile=""
+    if [ -h /etc/localtime ]; then
+        tzfile=$(readlink -f /etc/localtime)
     elif [ -f /usr/share/zoneinfo/UTC ]; then
-        cp -f /usr/share/zoneinfo/UTC $XBPS_MASTERDIR/etc/localtime
+        tzfile=/usr/share/zoneinfo/UTC
+    else
+        # Should never happen.
+        msg_warn "No local timezone configuration file created."
+    fi
+    # If tzfile exists, create parent directory, copy file and link to 
+    # /etc/localtime
+    if [ -f "$tzfile" ] ; then
+        mkdir -p $XBPS_MASTERDIR/$(dirname $tzfile)
+        cp /etc/localtime $XBPS_MASTERDIR/$tzfile
+        ln -sf ../$tzfile $XBPS_MASTERDIR/etc/localtime
```

Comment:
I would argue to always copy `/usr/share/zoneinfo/UTC` if it's exist to make the build maximum reproducible.

Reply:
I agree, but kept the original logic for a first draft. Not sure if there are any side effects I haven't thought of if localtime is always UTC. If there are no objections, I'll simplify the patch to always set localtime to UTC inside the chroot.